src/PdfReader/index.tsx

```diff
@@ -104,17 +104,25 @@ export default function usePdfReader(args: ReaderArguments): ReaderReturn {
       throw new Error('Manifest has no Reading Order');
     }
 
-    const resourceUrl = getResourceUrl(
+    const currentResource = getResourceUrl(
       state.resourceIndex,
       manifest.readingOrder
     );
-    loadResource(resourceUrl, proxyUrl).then((data) => {
-      dispatch({
-        type: 'RESOURCE_FETCH_SUCCESS',
-        resource: { data },
+
+    const fetchResource = async () => {
+      const resourceUrl = await getContent(currentResource);
+
+      loadResource(resourceUrl, proxyUrl).then((data) => {
+        dispatch({
+          type: 'RESOURCE_FETCH_SUCCESS',
+          resource: { data },
+        });
       });
-    });
-  }, [state.resourceIndex, manifest, proxyUrl]);
+    };
```

Comment:
Here is an example of how `getContent` will be used:
```
const getContent = async (linkUrl: string) => {
  const res = await fetch(linkUrl.toString(), {
    method: "GET",
    headers: {
      Authorization: `Bearer access_token`,
    },
  });
  return res.url;
};
// ...
<WebReader
  webpubManifestUrl={manifestUrl}
  proxyUrl={proxyUrl}
  pdfWorkerSrc={pdfWorkerSrc}
  headerLeft={<BackButton />}
  injectablesFixed={injectables}
  getContent={isLimitedAccess ? getContent : null}
/>
```
The link that will be passed to `getContent` would be a `/fulfill` endpoint that redirects to the returned resource URL.  This resource URL is then processed by the PDFReader's `loadResource`.

The default `getContent` (`fetchAsTxt`)  is meant to keep the functionality of `loadResource` as is. I could rename it to reflect its behavior.

Reply:
I see, so the contents of the books are actually publicly accessible on the web, but you just have to know the url to call, is that right? Couldn't a user see that url in their browser and share it then?

In any case, the API for `getContent` is that it is expected to return the resource itself, which you should be able to implement here simply enough by just fetching the url instead of returning it, and returning the Uint8Array from `getContent`. Does that make sense?

Reply:
The contents require authentication, but the url can be shared by the user post-auth. The backend team is aware of this and is looking at different solutions. 

Yes, I'll update the `getContent` definition for this.